raft/src/test/java/org/apache/kafka/raft/internals/KafkaRaftMetricsTest.java

```diff
@@ -63,7 +66,9 @@ public void tearDown() {
     private QuorumState buildQuorumState(Set<Integer> voters) {
         return new QuorumState(
             OptionalInt.of(localId),
-            voters,
+            localDirectoryId,
+            () -> VoterSetTest.voterSet(VoterSetTest.voterMap(voters, false)),
+            () -> (short) 0,
```

Comment:
I made the parameterized and configured them to run with `kraft.version` 0 and 1.

In general we didn't really need to test both cases because `kraft.version` affects the persisted state not the in-memory state. Since those tests are not recreating a `QuorumState` object they don't affect the directory id used.